utils/submissionDataFromGoogleMaps.ts

```diff
@@ -0,0 +1,124 @@
+import puppeteer from 'puppeteer'
+import dotenv from 'dotenv'
+
+//Load the api key for google maps
+dotenv.config()
+
+// This function ensures that we can redirect in a headerless browser using puppeteer to extract url once it has coordinates
+const extractRedirectedUrl = async (initialUrl: string): Promise<string> => {
+  const browser = await puppeteer.launch({ headless: true }) // ensures it doesn't open an actual browser
```

Comment:
Why is this step necessary? Can you add a JSDoc comment so there is a clear explanation. 

I assume it's because googlemapURLs on the phone are different and we want to get the 'long' url?

Reply:
Yea I can add it to the code. I have a comment at the top. Maybe it isn't clear. I wrote in the PR descriptions the thought process. But as an example. The url for a mobile url from Apple is something like https://maps.app.goo.gl/wQMavNqFCv3HVVqh7?g_st=ic

But we need to get the coordinates to use the logic below to use the Google places api in the most efficient manner with the data the user submits. 

This step ensures we get a consistent url with coordinates. Since the pattern on a browser is consistently like this

https://www.google.com/maps/place/McDonald's+Akasaka+Station/@35.6722412,139.7360999,16z/data=!3m1!4b1!4m6!3m5!1s0x60188b80c35827c9:0x70699f09e29c704!8m2!3d35.6722412!4d139.7360999!16s%2Fg%2F1tmkkzd9?entry=ttu